### PR TITLE
Initial Map Work

### PR DESCRIPTION
**Pull request by ewanmay**
- Created highlighting for individual countries
- Created legend
- Added zoom to bounds feature

### CODE REVIEW

**Comment by abeljoseph** [comment]
Please rebase

File: react-app/src/components/map/Map.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      color: 'white',
+      dashArray: '3',
+      fillOpacity: 0.7
+    }
+  }
+
+  const getBuckets = (features: GeoJSON.Feature[]) => {
+    var maxSeroprevalence = Math.max.apply(Math, features.filter(o => o.properties?.seroprevalence).map((o) => o?.properties?.seroprevalence));
+    var roundedMax = Math.ceil(maxSeroprevalence);
+    var step = parseFloat((roundedMax / 6).toFixed(1));
```
[the remark this answers is not in the document]
Comment (ewanmay) [reply react-app/src/components/map/Map.tsx]: I did it based off of the mock

File: react-app/src/components/map/Map.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      weight: 2,
+      opacity: 1,
+      color: 'white',
+      dashArray: '3',
+      fillOpacity: 0.7
+    }
+  }
+
+  const getBuckets = (features: GeoJSON.Feature[]) => {
+    var maxSeroprevalence = Math.max.apply(Math, features.filter(o => o.properties?.seroprevalence).map((o) => o?.properties?.seroprevalence));
```
[the remark this answers is not in the document]
Comment (ewanmay) [reply react-app/src/components/map/Map.tsx]: I should not C# and javascript

File: react-app/src/App.js [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import React from 'react';
 import { AppContextProvider } from "./context";
 import Component from "./components/Component"
+import Map from "./components/map/Map";
```
Comment (atatmaja) [inline react-app/src/App.js:4]: no need for this App.js anymore, replaced by App.tsx

File: src/react-app-env.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (atatmaja) [inline src/react-app-env.d.ts:1]: don't think we need to commit this, maybe add to gitignore

**Pull request merged**